(docs)troubleshooting - add JS error (#1948)
telerik/blazor-docs: troubleshooting/js-errors.md

## Object doesn't support property or method 'assign'

Under IE, you may get errors similar to `Object doesn't support property or method 'assign'` or errors that relate other modern JS features that are not supported under IE. The reason is that we use modern code that may not work under IE - it is not one of the [browsers we support]({%slug system-requirements%}#browser-support), and WebAssembly does not work on it anyway, so modern Blazor apps won't run on IE regardless.

> ## Maximum call stack size exceeed

Reviewer: ```suggestion
## Maximum call stack size exceeded
```

> Some common causes for the `Maximum call stack size exceeed` error are:

Reviewer: ```suggestion
Some common causes for the `Maximum call stack size exceeded` error are:
```

* Update from older to newer .NET version
* Old version of the `Telerik.UI.for.Blazor` (4.6.0 and older) with .NET 8
* All references to `Telerik.UI.for.Blazor` in your app are not for latest `Telerik.UI.for.Blazor` version 

> * The project is runnable in one browser and not in another

Reviewer: Refactor and reword this section. The main statement should be that an old `telerik-blazor.js` file (<= 4.5.0) is used in a .NET 8 app. The reason for the situation can be upgrade, cache, etc.

In addition, on Mac I can reproduce the problem with 4.5.0 in .NET 6 too. This implies that .NET is using the latest SDK to build the app, no matter the .NET version in the app. This is a rare case, as most of our customers are on Windows, but we can also mention it.

The solution is:

> 1. Close the opened project of the app from Visual Studio.

Reviewer: ```suggestion
1. Close Visual Studio.
```

> 1. Delete all the `bin` and `obj` files in all projects in the solution.

Reviewer: ```suggestion
1. Delete all `bin` and `obj` folders in all projects in the solution.
```

1. Clear the browser cache.
1. Reopen Visual Studio and rebuild the app. 

By clearing the browser cache, the `the telerik-blazor.js` file will reload and the error should go away.

You can also add a cache buster for your end users, which will have the same effect:

> `<script src="_content/Telerik.UI.for.Blazor/js/telerik-blazor.js?version501"></script>`

Reviewer: ```suggestion
`<script src="_content/Telerik.UI.for.Blazor/js/telerik-blazor.js?version{{site.uiForBlazorLatestVersion}}"></script>`
```

## Object doesn't support property or method 'assign'

Under IE, you may get errors similar to `Object doesn't support property or method 'assign'` or errors that relate other modern JS features that are not supported under IE. The reason is that we use modern code that may not work under IE - it is not one of the [browsers we support]({%slug system-requirements%}#browser-support), and WebAssembly does not work on it anyway, so modern Blazor apps won't run on IE regardless.

## Maximum call stack size exceeded

> The error indicates that a [.NET 8 app is using an old version (<= 4.5.0) of the `telerik-blazor.js` file]({%slug troubleshooting-kb-js-errors-maximum-call%}). If the `Telerik UI for Blazor` package version is up-to-date, a possible cause for the error is browser caching.

Reviewer: ```suggestion
The error indicates that a [.NET 8 app is using a `telerik-blazor.js` file that is for version `4.5.0` or older]({%slug common-kb-maximum-call-stack-exceeded%}). If the Telerik UI for Blazor package version is up-to-date, a possible cause for the error is browser caching.
```

> The error indicates that a [.NET 8 app is using a `telerik-blazor.js` file that is for version `4.5.0` or older]({%slug common-kb-maximum-call-stack-exceeded%}). If the Telerik UI for Blazor package version is up-to-date, a possible cause for the error is browser caching.

Reviewer 2: ```suggestion
The error indicates that a [.NET 8 app is using a `telerik-blazor.js` file that is for version `4.5.0` or earlier]({%slug common-kb-maximum-call-stack-exceeded%}). If the Telerik UI for Blazor package version is up-to-date, a possible cause for the error is browser caching.
```
